### PR TITLE
fix cos_scores_top_k_idx when using DRPES

### PR DESCRIPTION
**Pull request by NouamaneTazi**
Fixes https://github.com/beir-cellar/beir/issues/104

When gathering all results using `metric.compute` we sometimes mixed the batches ids. This PR aims to correct the documents ids when using DRPES.

Now DRPES gives correct results independently of how we set `corpus_chunk_size`
```python
>>> evaluation.run(model, output_folder=None, eval_splits=["test"], corpus_chunk_size=260)
Time taken to retrieve: 10.08 seconds
INFO:root:

INFO:root:NDCG@1: 0.2000
INFO:root:NDCG@3: 0.2567
INFO:root:NDCG@5: 0.2807
INFO:root:NDCG@10: 0.2953
INFO:root:NDCG@100: 0.3426
INFO:root:NDCG@1000: 0.3779
INFO:root:

INFO:root:MAP@1: 0.1867
INFO:root:MAP@3: 0.2366
INFO:root:MAP@5: 0.2505
INFO:root:MAP@10: 0.2574
INFO:root:MAP@100: 0.2660
INFO:root:MAP@1000: 0.2672
INFO:root:

INFO:root:Recall@1: 0.1867
INFO:root:Recall@3: 0.3019
INFO:root:Recall@5: 0.3581
INFO:root:Recall@10: 0.3994
INFO:root:Recall@100: 0.6336
INFO:root:Recall@1000: 0.9162
INFO:root:

INFO:root:P@1: 0.2000
INFO:root:P@3: 0.1078
INFO:root:P@5: 0.0787
INFO:root:P@10: 0.0443
INFO:root:P@100: 0.0070
INFO:root:P@1000: 0.0010
```

cc @Muennighoff @thakur-nandan

### CODE REVIEW

File: beir/retrieval/search/dense/exact_search_multi_gpu.py [hunk 1]
```diff
@@ -196,6 +193,9 @@ def _encode_multi_process_worker(self, process_id, device, model, input_queue, r
                     cos_scores_top_k_values = cos_scores_top_k_values.T.unsqueeze(0).detach()
                     cos_scores_top_k_idx = cos_scores_top_k_idx.T.unsqueeze(0).detach()
 
+                    # correct sentence ids
+                    cos_scores_top_k_idx = cos_scores_top_k_idx + id * self.corpus_chunk_size
```
Comment (Muennighoff) [inline beir/retrieval/search/dense/exact_search_multi_gpu.py:197]:
```suggestion
                    cos_scores_top_k_idx += id * self.corpus_chunk_size
```

**Review by Muennighoff — APPROVED** [review]
Amazing work! Confirming that I'm getting the same results now:

`DenseRetrievalParallelExactSearch`
```json
{
  "dataset_version": null,
  "mteb_version": "0.0.2",
  "test": {
    "evaluation_time": 189.38,
    "map_at_1": 0.41606,
    "map_at_10": 0.50777,
    "map_at_100": 0.51611,
    "map_at_1000": 0.51655,
    "map_at_3": 0.47967,
    "map_at_5": 0.49714,
    "ndcg_at_1": 0.44,
    "ndcg_at_10": 0.5538,
    "ndcg_at_100": 0.59487,
    "ndcg_at_1000": 0.60719,
    "ndcg_at_3": 0.50475,
    "ndcg_at_5": 0.5294,
    "precision_at_1": 0.44,
    "precision_at_10": 0.078,
    "precision_at_100": 0.01003,
    "precision_at_1000": 0.00111,
    "precision_at_3": 0.20333,
    "precision_at_5": 0.13933,
    "recall_at_1": 0.41606,
    "recall_at_10": 0.68456,
    "recall_at_100": 0.881,
    "recall_at_1000": 0.98,
    "recall_at_3": 0.55044,
    "recall_at_5": 0.61194
  }
}
```

`DenseRetrievalExactSearch`
```json
{
  "test": {
    "evaluation_time": 787.86,
    "map_at_1": 0.41606,
    "map_at_10": 0.50777,
    "map_at_100": 0.51611,
    "map_at_1000": 0.51655,
    "map_at_3": 0.47967,
    "map_at_5": 0.49714,
    "ndcg_at_1": 0.44,
    "ndcg_at_10": 0.5538,
    "ndcg_at_100": 0.59487,
    "ndcg_at_1000": 0.60719,
    "ndcg_at_3": 0.50475,
    "ndcg_at_5": 0.5294,
    "precision_at_1": 0.44,
    "precision_at_10": 0.078,
    "precision_at_100": 0.01003,
    "precision_at_1000": 0.00111,
    "precision_at_3": 0.20333,
    "precision_at_5": 0.13933,
    "recall_at_1": 0.41606,
    "recall_at_10": 0.68456,
    "recall_at_100": 0.881,
    "recall_at_1000": 0.98,
    "recall_at_3": 0.55044,
    "recall_at_5": 0.61194
  }
}
```
